### PR TITLE
Jotform processing

### PR DESCRIPTION
**Pull request by bencodeorg**
This PR has a couple of aims:
1. migrations to create `survey_questions` and `survey_answers` tables
2. reshapes Jotform survey answers (from PdWorkshopDailySurveys) and Jotform surveys (from PdSurveyQuestions) to one row per question answer and one row per question, respectively. 
3. schedules cron to drop and regenerate tables

To test, I imported data from both of these tables into my local DB. Reshaping `PdWorkshopDailySurveys` produces 1,000,000+ rows in the `survey_answers` table, so one question here would be around scalability of this approach.

It follows an approach that has been used in the past to reshape data for analysis:
https://github.com/code-dot-org/code-dot-org/blob/staging/bin/cron/create_rollup_tables

Ideally, this structure would allow us to append additional surveys and their responses.

### CODE REVIEW

**Comment by bencodeorg** [comment]
@hacodeorg -- could you look at this and see whether there is overlap between what I have here and  the work you are doing to make changes to the Jotform processing pipeline? Also may be easier to go over together.

Also, my intention here is that we could append on survey results from other surveys (not just `PdWorkshopDailySurvey`). Where are Jotform surveys from the upcoming facilitator spring training going to be stored, and will it follow a similar structure to `PdWorkshopDailySurvey`?

**Comment by hacodeorg** [comment]
It will not overlap with my other work [here](https://github.com/code-dot-org/code-dot-org/blob/87a4318128716d27bbf0b79ca7d0f61db3b1e666/docs/plc/survey-summary-design.md), which focus on summarizing and presenting survey results in workshop dashboard. We've decided to implement a new pipeline parallel with the existing one (and eventually replace it way later).

I'm not aware of any JotForm survey for facilitator training (I assume this is the FiT weekend workshop). If that is a new survey we want to add this year, it'll likely be handled by the new pipeline and will be stored in a different format. (You can see a high level view of the new database format in the diagram in section 4.3.B in my design doc.)

We can talk more about this tomorrow.

> @hacodeorg -- could you look at this and see whether there is overlap between what I have here and the work you are doing to make changes to the Jotform processing pipeline? Also may be easier to go over together.
> 
> Also, my intention here is that we could append on survey results from other surveys (not just `PdWorkshopDailySurvey`). Where are Jotform surveys from the upcoming facilitator spring training going to be stored, and will it follow a similar structure to `PdWorkshopDailySurvey`?

File: dashboard/db/schema.rb [hunk 1]
```diff
@@ -1338,6 +1338,29 @@
...
+    t.integer  "user_id"
+    t.bigint   "form_id"
+    t.bigint   "submission_id"
+    t.string   "question_id"
+    t.text     "answer_value",  limit: 65535
+    t.datetime "created_at",                  null: false
+    t.datetime "updated_at",                  null: false
+  end
+
+  create_table "survey_questions", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci" do |t|
```
Comment (hacodeorg) [inline dashboard/db/schema.rb:1351]: I understand that this is our current design in which we separate submissions (with answers) from questions. My main concern is answers are fixed while questions could change over time. An answer submitted a month ago might be for a different question than the one we download today even though they have the same question id.

**Comment by hacodeorg** [comment]
Is this work only for analysis purpose or more?

**Review by hacodeorg — APPROVED** [review]
Discussed offline, this PR looks good to me.

**Pull request merged**